### PR TITLE
DOC: Use official MATLAB spelling in numpy-for-matlab-users.rst

### PR DESCRIPTION
**Pull request by cooperrc**
There is a current PR #17159, where we (@eric-wieser @melissawm @bjnath @seberg) are updating the numpy-for-matlab-users.rst guide. 

[Comment](https://github.com/numpy/numpy/pull/17159#discussion_r481107497) suggested updating the copytright (R) symbols as per MATLAB's suggested use:

@bjnath:
> MathWorks says

>> use the relevant trademark symbol to mark the first occurrences

> and

>> The word MATLAB®should appear in all capital letters.

### CODE REVIEW

File: doc/source/user/numpy-for-matlab-users.rst [hunk 1]
```diff
@@ -53,7 +53,7 @@ Some Key Differences
        things that Matlab just cannot, for instance dealing properly with
```
Comment (eric-wieser) [inline doc/source/user/numpy-for-matlab-users.rst:53]:
```suggestion
       things that MATLAB just cannot, for instance dealing properly with
```
And throughout the file.

**Review by eric-wieser — APPROVED** [review]

File: doc/source/user/numpy-for-matlab-users.rst (hunk 1 above)
Comment (melissawm) [reply doc/source/user/numpy-for-matlab-users.rst:53]: Keep in mind there are also a couple of lowercase "matlab" instances that should be fixed.

**Review by melissawm — CHANGES_REQUESTED** [review]

**Comment by cooperrc** [comment]
@melissawm and @eric-wieser that was a good idea to catch the case as well as the (R) symbols. 

Current commit 233d2816b8852f86fd774035e1c8f42c4e7a25a9 updates all Matlab and matlab to MATLAB

**Review by melissawm — APPROVED** [review]
Thanks, @cooperrc ! I'm just not sure what's going on with the CI...

File: doc/source/user/numpy-for-matlab-users.rst [hunk 2]
```diff
@@ -712,10 +712,10 @@ this is just an example, not a statement of "best practices"):
...
 
-See http://mathesaurus.sf.net/ for another MATLAB®/NumPy
+See http://mathesaurus.sf.net/ for another MATLAB/NumPy
 cross-reference.
 
 An extensive list of tools for scientific work with python can be
 found in the `topical software page <https://scipy.org/topical-software.html>`__.
 
-MATLAB® and SimuLink® are registered trademarks of The MathWorks.
+MATLAB and SimuLink® are registered trademarks of The MathWorks.
```
Comment (eric-wieser) [inline doc/source/user/numpy-for-matlab-users.rst:721]: Would be inclined to leave this one alone,
```suggestion
MATLAB® and SimuLink® are registered trademarks of The MathWorks.
```

**Comment by cooperrc** [comment]
> Thanks, @cooperrc ! I'm just not sure what's going on with the CI...

I'm not sure either. Its odd that the checks were having trouble.

**Comment by seberg** [comment]
Thanks Ryan, circleci is happy, so all good!

**Comment by bjnath** [comment]
Will there now be merge conflicts with #17159, where many changed lines include the 'matlab' string in its various forms?

**Comment by seberg** [comment]
Oh, yeah :(, that is probably an annoying merge conflict there

**Comment by eric-wieser** [comment]
It should be an easy conflict to solve, since @cooperrc authored both PRs. Keeping the version of the file as it is in #17159 should work just fine.

**Comment by eric-wieser** [comment]
`git rebase -Xtheirs origin/master` should do the trick there

**Comment by cooperrc** [comment]
> `git rebase -Xtheirs origin/master` should do the trick there

@eric-wieser, should I run this, then push the changes?

**Comment by eric-wieser** [comment]
Yes, I think so.You'll I'd recommend running `git tag just-in-case-i-break-it` before hand, for eponymous reasons. So overally
```
git pull origin  # get any changes from the other PR made in the web version
git fetch upstream  # or whatever you called this repo
git rebase -Xtheirs upstream/master
git push origin -f
```

**Comment by cooperrc** [comment]
OK, its been done. First glance looks ok. I was nervous with the `-f` option in a `push`. 

Thanks @eric-wieser!

**Pull request merged**